.github/workflows/coverage.yml

```diff
@@ -54,16 +55,24 @@ jobs:
           python -m pip install -c constraints.txt -r requirements-dev.txt -r requirements-optional.txt
           python tools/report_numpy_state.py
           stestr run
-          # We set the --source-dir to '.' because we want all paths to appear relative to the repo
-          # root (we need to combine them with the Python ones), but we only care about `grcov`
-          # keeping the `crates/*` files; we don't care about coverage in dependencies.
-          grcov . --binary-path target/debug/ --source-dir . --output-type lcov --output-path rust.info --llvm --branch --parallel --keep-only 'crates/*'
         env:
           QISKIT_TEST_CAPTURE_STREAMS: 1
           QISKIT_PARALLEL: FALSE
           QISKIT_IGNORE_USER_SETTINGS: TRUE
           PYTHON: "coverage run --source qiskit --parallel-mode"
 
+      - name: Generate ctest coverage report
+        run: |
+          make ctest
+          # We set the --source-dir to '.' because we want all paths to appear relative to the repo
+          # root (we need to combine them with the Python ones), but we only care about `grcov`
+          # keeping the `crates/*` files; we don't care about coverage in dependencies.
+          grcov . --binary-path target/debug/ --source-dir . --output-type lcov --output-path rust.info --llvm --branch --parallel --keep-only 'crates/*'
+        env:
+          CARGO_INCREMENTAL: 0
+          RUSTFLAGS: "-Cinstrument-coverage"
+          LLVM_PROFILE_FILE: "qiskit-%p-%m.profraw"
+
```

Comment:
Might be clearer to separate out the `grcov` into its own separate step now, and make sure the `name` fields on the `unittest` and `ctest` jobs both make sense - it's starting to get a bit muddled, and a bit magic that `grcov` runs in one of two test steps.